src/app.ts

```diff
@@ -13,15 +13,86 @@ interface TemperatureSummary {
   average: number
 }
 
+const allReadings = [
+	{
+		time: new Date('1/1/2021'),
+		temperature: 10,
+		city: 'Utah',
+	},
+	{
+		time: new Date('1/1/2021'),
+		temperature: 9,
+		city: 'Utah',
+	},
+	{
+		time: new Date('1/1/2021'),
+		temperature: 11,
+		city: 'Utah',
+	},
+	{
+		time: new Date('1/1/2021'),
+		temperature: 3,
+		city: 'New York',
+	},
+	{
+		time: new Date('1/1/2021'),
+		temperature: 2,
+		city: 'New York',
+	},
+	{
+		time: new Date('1/1/2021'),
+		temperature: 7,
+		city: 'New York',
+	},
+];
+
+
+class Summary implements TemperatureSummary {
+  constructor (public first:number, public last: number, public high:number, public low:number, public average:number) {
+  }
+};
+
+let processedReadings: TemperatureSummary;
+
 function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+// add here your code
+ const listByTime= readings.sort((a, b) => (a.time > b.time) ? 1 : -1)
```

Comment:
you don't need to sort the dates

Reply:
because it said to return the 1st and last reading of the day, I sorted by time. Is this not right?